flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/PlannerBase.scala

```diff
@@ -173,7 +165,12 @@ abstract class PlannerBase(
       case catalogSink: CatalogSinkModifyOperation =>
         val input = getRelBuilder.queryOperation(modifyOperation.getChild).build()
         getTableSink(catalogSink.getTablePath).map(sink => {
-          TableSinkUtils.validateSink(catalogSink.getChild, catalogSink.getTablePath, sink)
+          TableSinkUtils.validateSink(catalogSink, catalogSink.getTablePath, sink)
+          sink match {
+            case partitionableSink: PartitionableTableSink =>
```

Comment:
Only set static partitions when sink has partitioned fields, i.e. is a partitioned sink. Currently, we set a partition map always which may have unexpected effects.

Reply:
The `PartitionableTableSink` is the only trait that `Flink` recognize as a partiton sink, if the partitioned fields is empty, the static partitions are always empty(we have checked this in sql node parse). So what exactly is the unexpected effects ?

BTW, i don't think the two methods `#getPartitionFieldNames` and `#setStaticPartition` should affect each other.

Reply:
From user's point of view, if it is not a partitioned sink (`getPartitionFieldNames()` returns empty), the `setStaticPartition` should never be called. However, currently, `setStaticPartition` is always be called (may pass in an empty partition map).

Reply:
So what's the problem here ? Isn't the empty static partitions the right thing here ?